### PR TITLE
Transition testing framework to pytest

### PR DESCRIPTION
**Pull request by jadchaar**
## Pull Request Checklist

Thank you for taking the time to improve Arrow! Before submitting your pull request, please check all *appropriate* boxes:

- [x] 🧪 Added **tests** for changed code.
- [x] 🛠️ All tests **pass** when run locally (run `tox` or `make test` to find out!).
- [x] 📚 Updated **documentation** for changed code.
- [x] ⏩ Code is **up-to-date** with the `master` branch.

If you have *any* questions about your code changes or any of the points above, please submit your questions along with the pull request and we will try our best to help!

## Description of Changes
Closes: https://github.com/crsmithdev/arrow/issues/594

This is the first iteration of the transition to Pytest. We should ultimately move away from the x-unit style of `setUp()` functions to native Pytest fixtures, but this will do for now.

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -1,10 +1,9 @@
 backports.functools_lru_cache==1.6.1; python_version == "2.7"
-chai==1.1.2
 dateparser==0.7.*
-mock==3.0.*
-nose==1.3.7
-nose-cov==1.6
 pre-commit==1.20.*
+pytest
+pytest-cov
+pytest-mock
```
Comment (systemcatch) [inline requirements.txt]: We should pin all of these for better reproducibility.

**Comment by systemcatch** [comment]
Hey @jadchaar I was able to successfully clone your branch and run the new tests, everything passed fine.

**Comment by jadchaar** [comment]
Fantastic. I have locked the dependency versions like you requested.

**Comment by systemcatch** [comment]
Looks like the last version of pytest to support 2.7 is 4.6 (https://docs.pytest.org/en/latest/py27-py34-deprecation.html). 😞

**Comment by jadchaar** [comment]
@systemcatch I pushed a fix for this in the latest commit.

**Comment by jadchaar** [comment]
I don't have anything to add :). I'll go ahead and merge this.

**Pull request merged**